### PR TITLE
Prow: Upgrade cluster to v1.26.4 and document

### PR DESCRIPTION
**Pull request by lentzi90**
- Add documentation for how to upgrade the Kubernetes cluster where Prow is running.
- Add documentation for how to replace or perform maintenance on the bastion host for the Prow cluster.
- Upgrade the Kubernetes cluster to v1.26.4.

### CODE REVIEW

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

Very good documentation.

**Review by tuminoid — APPROVED** [review]
/lgtm

File: prow/README.md [hunk 1]
```diff
@@ -436,3 +436,135 @@ In any case we should follow the steps below to enable prow:
...
+Or perhaps someone did some changes in the cluster without committing them in
+the manifests.
+A simple `kubectl diff` will save you from surprises many times.
+
+### Upgrading Kubernetes and node images
+
+When it is time to upgrade to a new Kubernetes version (or just update the node
+image), you will first need to build a new node image (see section above).
+After this you need to create new OpenStackMachineTemplates for the image and
+switch the KubeadmControlPland and MachineDeployment to these new template.
```
Comment (kashifest) [inline prow/README.md:466]:
```suggestion
switch the KubeadmControlPlane and MachineDeployment to these new template.
```

File: prow/README.md [hunk 2]
```diff
@@ -436,3 +436,135 @@ In any case we should follow the steps below to enable prow:
...
+If the Kubernetes version changed you also need to update the Kubernetes
```
Comment (kashifest) [inline prow/README.md:467]:
```suggestion
If the Kubernetes version has changed you also need to update the Kubernetes
```

**Review by kashifest — APPROVED** [review]
Small nits.
/hold

**Comment by lentzi90** [comment]
Thanks @kashifest , fixed!

**Comment by kashifest** [comment]
/lgtm cancel 
/hold cancel
@Sunnatillo @tuminoid PTAL

**Pull request merged**